qiskit/test/mock/utils/json_decoder.py

```diff
@@ -0,0 +1,50 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Utils to decode fake backend configurations from json
+"""
+
+from typing import Dict, Union, List
+
+
+def decode_backend_configuration(config: Dict) -> None:
```

Comment:
It seems the accompanying code for the defaults/properties are missing (https://github.com/Qiskit/qiskit-ibmq-provider/blob/c4738b52fd645b65dec38c7661ece2a7477c44a2/qiskit/providers/ibmq/utils/json_decoder.py#L36-L65)?

Reply:
> It seems the accompanying code for the defaults/properties are missing (https://github.com/Qiskit/qiskit-ibmq-provider/blob/c4738b52fd645b65dec38c7661ece2a7477c44a2/qiskit/providers/ibmq/utils/json_decoder.py#L36-L65)?

I've added them and used at the appropriate places.